applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_cr_beam_element_3D2N.h

```diff
@@ -0,0 +1,77 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:     BSD License
+//           license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Martin Fusseder, https://github.com/MFusseder
+//
+
+
+#if !defined(ADJOINT_FINITE_DIFFERENCE_CR_BEAM_ELEMENT_H_INCLUDED )
+#define  ADJOINT_FINITE_DIFFERENCE_CR_BEAM_ELEMENT_H_INCLUDED
+
+#include "adjoint_finite_difference_base_element.h"
+
+namespace Kratos
+{
+
+class AdjointFiniteDifferenceCrBeamElement : public AdjointFiniteDifferencingBaseElement
+{
+public:
+    KRATOS_CLASS_POINTER_DEFINITION(AdjointFiniteDifferenceCrBeamElement);
+
+    typedef Element BaseType;
+    typedef BaseType::GeometryType GeometryType;
+    typedef BaseType::NodesArrayType NodesArrayType;
+    typedef BaseType::PropertiesType PropertiesType;
+    typedef BaseType::IndexType IndexType;
+    typedef BaseType::SizeType SizeType;
+    typedef BaseType::MatrixType MatrixType;
+    typedef BaseType::VectorType VectorType;
+    typedef BaseType::EquationIdVectorType EquationIdVectorType;
+    typedef BaseType::DofsVectorType DofsVectorType;
+
+    AdjointFiniteDifferenceCrBeamElement(): AdjointFiniteDifferencingBaseElement()
+    {
+    }
+
+    AdjointFiniteDifferenceCrBeamElement(Element::Pointer pPrimalElement);
+
+    ~AdjointFiniteDifferenceCrBeamElement() override;
+
+    BaseType::Pointer Create(Element::Pointer pPrimalElement) const override;
```

Comment:
I'm not sure how you intend to use these Create methods, but it might be simpler in this case to just have the constructor. The Create is useful if you want a factory through some base interface, but this is not part of the standard Element interface so i'm not sure how useful it will be.

Reply:
I see, this is unused. 
I could think of a Create method that overwrites the base class method and internally creates a primal element, to directly store it inside the adjoint element.  

```cpp
    virtual Pointer Create(IndexType NewId, NodesArrayType const& ThisNodes,
                           PropertiesType::Pointer pProperties) const
    {
        p_primal_element =  Kratos::make_shared<PrimalElement>(NewId, GetGeometry().Create(ThisNodes), pProperties);
        return Kratos::make_shared<AdjointElement>(p_primal_element);
    }
```

Like this we could create these adjoint elements more easily for e.g. testing

Reply:
Removed the Create Methods for now